### ISSUE TITLE
[Feature/Improvement] Confirmation on workout deletion

### ISSUE THREAD

**Issue by a-mabe** [a-mabe/OpenHIIT#104]
**Purpose:** As a user, I would like to be asked for confirmation before a workout is deleted.

**More details:** Some more details on the feature/improvement

Currently, it is easy to accidentally delete a workout as workouts are immediately deleted upon tapping the delete icon.

Implement something similar to the alert dialog shown here: https://medium.com/multiverse-software/alert-dialog-and-confirmation-dialog-in-flutter-8d8c160f4095

**Acceptance Criteria:**

- Deletion confirmation screen appears after tapping the delete icon.

**Comment by gantanikhilraj** [comment]
Hi @a-mabe ,I would like to work on this, Assign this to me.

**Comment by a-mabe** [comment]
Hello @gantanikhilraj, thank you for contributing to the project, I will assign the issue to you

**Comment by gantanikhilraj** [comment]
@a-mabe Can you provide some more information about the issue ?

**Comment by a-mabe** [comment]
Sure, there is a delete button that is accessible after tapping on a timer, shown in the screenshot below:

<img width="394" alt="Screenshot 2023-11-25 at 2 37 30 PM" src="https://github.com/a-mabe/OpenHIIT/assets/39250511/24a4dbff-0017-4d21-9519-2a02521ab4a7">

When the user taps this button, the timer is immediately deleted. This makes it easily to accidentally tap the icon and delete the timer without intending to.

Instead of immediately deleting the timer, there should be a pop-up that asks for confirmation to delete the timer. This way, on an accidental tap of the delete button, the user is able to undo their action and keep the timer.

The pop-up can be something like the [AlertDialogue](https://api.flutter.dev/flutter/material/AlertDialog-class.html) shown in this blog post: https://medium.com/multiverse-software/alert-dialog-and-confirmation-dialog-in-flutter-8d8c160f4095

Something like:
```
Are you sure you would like to delete <timer name>?

Cancel        Delete
```

**Comment by a-mabe** [comment]
Thank you for the contribution @gantanikhilraj! Closed in #107
